lib/modules/datasource/typst/index.ts

```diff
@@ -0,0 +1,104 @@
+import { z } from 'zod';
+import { logger } from '../../../logger';
+import { cache } from '../../../util/cache/package/decorator';
+import { PackageHttpCacheProvider } from '../../../util/http/cache/package-http-cache-provider';
+import { GithubHttp } from '../../../util/http/github';
+import { LooseArray, Toml } from '../../../util/schema-utils';
+import { fromBase64 } from '../../../util/string';
+import { api as semver } from '../../versioning/semver-coerced';
+import { Datasource } from '../datasource';
+import type { GetReleasesConfig, Release, ReleaseResult } from '../types';
+
+const Versions = LooseArray(
+  z
+    .object({
+      type: z.literal('dir'),
+      name: z.string(),
+    })
+    .transform(({ name }) => name)
+    .refine((version) => semver.isValid(version)),
+);
+
+const SourceUrl = z
+  .object({ content: z.string() })
+  .transform(({ content }) => fromBase64(content))
+  .pipe(Toml)
+  .pipe(
+    z
+      .object({
+        package: z.object({
+          repository: z.string(),
+        }),
+      })
+      .transform((data) => data.package.repository),
+  )
+  .optional()
+  .catch(undefined);
+
+export class TypstDatasource extends Datasource {
+  static readonly id = 'typst';
+
+  override readonly defaultRegistryUrls = ['https://github.com'];
+
+  githubHttp: GithubHttp;
+
+  constructor() {
+    super(TypstDatasource.id);
+    this.githubHttp = new GithubHttp(TypstDatasource.id, {
+      baseUrl: 'https://api.github.com',
```

Comment:
add a readme that a token is required to get this datasource working when platform isn't github